### PR TITLE
feat(github): Review the PR template

### PR DESCRIPTION
**Pull request by chouetz**
### What does this PR do?
Change the PR template.

### Motivation
There are a lot of comments in the current template and several of them seem unnecessary to me (they just describe with more words the current section name). It is a drawback when updating the PR description as we are polluted with this comments.

### Additional Notes
I took the opportunity to reschuffle the sections, setting `test` and `drawbacks/trade-offs` sections before the `additional notes` which is an `appendix` section according to me (and I let a comment for this one only)

### CODE REVIEW

**Review by pducolin — APPROVED** [review]

**Review by CelianR — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,46 +1,25 @@
...
-* New contributors are highly encouraged to read our
-  [CONTRIBUTING](/CONTRIBUTING.md) documentation.
+* Contributors are encouraged to read our [CONTRIBUTING](/CONTRIBUTING.md) documentation.
 * Both Contributor and Reviewer Checklists are available at https://datadoghq.dev/datadog-agent/guidelines/contributing/#pull-requests.
 * The pull request:
   * Should only fix one issue or add one feature at a time.
   * Must update the test suite for the relevant functionality.
   * Should pass all status checks before being reviewed or merged.
 * Commit titles should be prefixed with general area of pull request's change.
+* Please fill below sections if possible with any relevant information or link.
```
Comment (KSerrania) [inline .github/PULL_REQUEST_TEMPLATE.md:9]:
```suggestion
* Please fill the below sections if possible with relevant information or links.
```

**Review by spikat — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,46 +1,25 @@
...
-  opening multiple pull requests instead of a single one.
--->
-
 ### Motivation
 
-<!--
-* What inspired you to submit this pull request?
-* Link any related GitHub issues or PRs here.
--->
+### Describe how to test/QA your changes
```
Comment (pgimalac) [inline .github/PULL_REQUEST_TEMPLATE.md:15]: I feel like having QA instructions at the end was convenient as reviewers don't need to check them, while they do want to look at the additional notes, the new layout means you have to ignore the middle of the description when reviewing (and similarly for QA-ers, you usually mostly look the QA section, so now you need to find it in the middle of the description)

Comment (chouetz) [reply .github/PULL_REQUEST_TEMPLATE.md:15]: I would probably argue that the way you test things are quite valuable for a reviewer as well:
- as we want to enrich our test coverage, reviewer can encourage tests
- it's an opportunity to try assessing if tests are relevant.

**Pull request merged**